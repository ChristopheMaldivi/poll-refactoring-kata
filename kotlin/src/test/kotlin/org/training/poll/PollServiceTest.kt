package org.training.poll

import org.approvaltests.Approvals
import org.junit.jupiter.api.AfterEach
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import java.util.*
import java.io.PrintStream
import java.io.ByteArrayOutputStream



class PollServiceTest {

    private val outContent = ByteArrayOutputStream()
    private val errContent = ByteArrayOutputStream()
    private val originalOut = System.out
    private val originalErr = System.err

    private val pollService = PollService()

    @BeforeEach
    fun setup() {
        System.setOut(PrintStream(outContent))
        System.setErr(PrintStream(errContent))
    }

    @AfterEach
    fun restoreStreams() {
        System.setOut(originalOut)
        System.setErr(originalErr)
    }

    @Test
    fun `test approval`() {
        // given
        pollService.createPoll(
            0,
            "What is your favorite color?",
            ArrayList(listOf("Blue", "Green", "Red", "Yellow"))
        )

        // when
        pollService.vote(0, 0)
        pollService.vote(0, 0)
        pollService.vote(0, 0)
        pollService.vote(0, 3)
        pollService.vote(0, 2)
        pollService.printResults(0)
        val output = outContent.toString()

        // then
        Approvals.verifyAll("", arrayOf(output))
    }
}