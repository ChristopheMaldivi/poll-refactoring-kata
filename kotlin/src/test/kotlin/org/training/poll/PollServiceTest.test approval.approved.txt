[0] = What is your favorite color?
Blue (3)
Green (0)
Red (1)
Yellow (1)

